internal/eventstore/repository/sql/query.go

```diff
@@ -243,14 +243,11 @@ func eventsScanner(useV1 bool) func(scanner scan, dest interface{}) (err error)
 	}
 }
 
-func prepareConditions(criteria querier, query *repository.SearchQuery, useV1 bool) (_ string, args []any) {
-	clauses, args := prepareQuery(criteria, useV1, query.InstanceID, query.InstanceIDs, query.ExcludedInstances)
-	if clauses != "" && len(query.SubQueries) > 0 {
-		clauses += " AND "
-	}
+func prepareConditions(criteria querier, query *repository.SearchQuery, useV1 bool) (clauses string, args []any) {
 	subClauses := make([]string, len(query.SubQueries))
 	for i, filters := range query.SubQueries {
 		var subArgs []any
+		filters = append([]*repository.Filter{query.InstanceID, query.InstanceIDs, query.ExcludedInstances}, filters...)
```

Comment:
If I understand correctly you add the filters for InstanceID, InstanceIDs and ExcludedInstances to each subquery instead of the main query?